python_ta/cfg/visitor.py

```diff
@@ -58,3 +65,49 @@ def visit_if(self, node: astroid.If) -> None:
         self.cfg.link_or_merge(end_else, after_if_block)
 
         self._current_block = after_if_block
+
+    def visit_while(self, node: astroid.While) -> None:
+        old_curr = self._current_block
+
+        # Handle "test" block
+        test_block = self.cfg.create_block()
+        test_block.add_statement(node.test)
+        self.cfg.link_or_merge(old_curr, test_block)
+
+        after_while_block = self.cfg.create_block()
+
+        # add while scope
+        self._scope.append(('while', test_block, after_while_block))
+
+        # Handle "body" branch
+        body_block = self.cfg.create_block(test_block)
+        self._current_block = body_block
+        for child in node.body:
+            child.accept(self)
+        end_body = self._current_block
+        self.cfg.link_or_merge(end_body, test_block)
+
+        # step outside loop
+        self._scope.pop()
+
+        # Handle "else" branch
+        else_block = self.cfg.create_block(test_block)
+        self._current_block = else_block
+        for child in node.orelse:
+            child.accept(self)
+        end_else = self._current_block
+
+        self.cfg.link_or_merge(end_else, after_while_block)
+        self._current_block = after_while_block
+
+    def visit_break(self, node: astroid.Break) -> None:
+        old_curr = self._current_block
+        old_curr.add_statement(node)
+        loop_scope = None
+        for scope in self._scope:
+            if scope[0] in ['while', 'for']:
+                loop_scope = scope
+        if not loop_scope:
+            raise SyntaxError('\'break\' outside loop')
+        self.cfg.link_or_merge(old_curr, loop_scope[2])
+        self._current_block.jump = node
```

Comment:
This line should be removed (see above comments about `jump`).

Also, I actually want to preserve any statements after the `break` in a separate (unreachable) node. This is actually useful to capture for later to represent exactly which statements are unreachable. You can support this by initializing a new empty node right at the end of this method, and setting `self._current_block` to that new empty node.